### PR TITLE
Testing withClass with forwardRef -  RSC-622

### PR DESCRIPTION
**Pull request by entroform**
Adding forwardRef to withClass helper hoc.
This also fixes error with OverflowTooltip not receiving ref from NxList.Item, NxList.Text, etc...

### CODE REVIEW

File: lib/src/util/withClass.tsx [hunk 1]
```diff
@@ -13,7 +13,20 @@ export default function withClass<E extends keyof JSX.IntrinsicElements>(
 ): FunctionComponent<JSX.IntrinsicElements[E]> {
   return function WithClassWrapper({ className, ...otherProps }: JSX.IntrinsicElements[E]) {
     const classes = classnames(withClassName, className);
-
     return React.createElement(El, { className: classes, ...otherProps});
   };
 }
+
+export function withClassAndForwardedRef<E extends keyof JSX.IntrinsicElements, R extends HTMLElement = HTMLElement>(
```
Comment (entroform) [inline lib/src/util/withClass.tsx:20]: Let me know what your thoughts are on this. I'm still trying to figure out how to derive specific subset of HTMLElement interface JSX.IntrinsicElements key, in the meantime for testing I just used HTMLElement

File: lib/src/util/withClass.tsx [hunk 2]
```diff
@@ -4,16 +4,30 @@
...
+type NativeElTypeFromDetailedHTMLProps<E extends keyof JSX.IntrinsicElements> =
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  JSX.IntrinsicElements[E] extends DetailedHTMLProps<any, infer A> ? A : never;
+
+type NativeElTypeFromSVGProps<E extends keyof JSX.IntrinsicElements> =
+  JSX.IntrinsicElements[E] extends SVGProps<infer A> ? A : never;
+
+type NativeElType<E extends keyof JSX.IntrinsicElements> =
+  NativeElTypeFromDetailedHTMLProps<E> extends never
+    ? NativeElTypeFromSVGProps<E> : NativeElTypeFromDetailedHTMLProps<E>;
```
Comment (rpokorny) [inline lib/src/util/withClass.tsx:19]: Was it necessary to have it be three types like this? I would have expected you to check against both `DetailedHTMLProps` and `SVGProps` in a single nested ternary, which would probably be more straightforward

**Comment by rpokorny** [comment]
Great work! Needs tests though

**Comment by entroform** [comment]
> Great work! Needs tests though

Yay thank you so much for your guidance: Added test here: 52b176d42882a60d1724d770baac1a8937c3b70c

File: lib/src/util/__tests__/withClass.test.tsx [hunk 3]
```diff
@@ -4,13 +4,26 @@
...
+import React from 'react';
+import { mount } from 'enzyme';
 import withClass from '../withClass';
 import { getShallowComponent } from '../../__testutils__/enzymeUtils';
 
 describe('withClass', function() {
   const ExampleComponent = withClass('hgroup', 'foo-bar'),
       getShallow = getShallowComponent(ExampleComponent, {});
 
+  it('forwards a ref to the specified label element', function() {
```
Comment (rpokorny) [inline lib/src/util/__tests__/withClass.test.tsx:16]:
```suggestion
  it('forwards a ref to the element', function() {
```

File: lib/src/util/withClass.tsx [hunk 4]
```diff
@@ -4,16 +4,27 @@
...
+import React, { forwardRef, DetailedHTMLProps, SVGProps } from 'react';
 import classnames from 'classnames';
 
+type NativeElType<E extends keyof JSX.IntrinsicElements> =
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  JSX.IntrinsicElements[E] extends DetailedHTMLProps<any, infer A>
+    ? A
+    : JSX.IntrinsicElements[E] extends SVGProps<infer B>
+      ? B
+      : never;
```
Comment (rpokorny) [inline lib/src/util/withClass.tsx:16]: I prefer that multi-line ternaries be formatted like this for readability:

```suggestion
  JSX.IntrinsicElements[E] extends DetailedHTMLProps<any, infer A> ? A : 
  JSX.IntrinsicElements[E] extends SVGProps<infer B> ? B : 
  never;
```

**Review by rpokorny — APPROVED** [review]

**Pull request merged**